src/main/java/com/books/storage/abstracts/UserDAO.java

```diff
@@ -0,0 +1,8 @@
+package com.books.storage.abstracts;
+
+import com.books.entities.User;
+
+public interface UserDAO {
```

Comment:
Do not mix concepts. If you want to follow "dao" naming conventions, so please keep all corresponded classes with dao suffix, if you prefer "repository" - use repository suffix